WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -201,8 +205,11 @@ extension DashboardViewController {
 private extension DashboardViewController {
 
     @objc func settingsTapped() {
+        guard let settingsViewController = UIStoryboard.dashboard.instantiateViewController(ofClass: SettingsViewController.self) else {
+            fatalError("Cannot instantiate `SettingsViewController` from Dashboard storyboard")
+        }
         ServiceLocator.analytics.track(.settingsTapped)
-        performSegue(withIdentifier: Segues.settingsSegue, sender: nil)
+        navigationController?.show(settingsViewController, sender: self)
```

Comment:
I think you can dow `self.show(settingsViewController, sender: self)` and `UIKit` will find the active navigation controller to send the message to!

Reply:
nice I like it! updated in https://github.com/woocommerce/woocommerce-ios/pull/2819/commits/473876d385bfe282fe0d24d4027cb4611f7fc3af